PSReadLine/PlatformWindows.cs

```diff
@@ -79,6 +79,21 @@ IntPtr templateFileWin32Handle
     [DllImport("kernel32.dll", CharSet = CharSet.Unicode, SetLastError = true)]
     internal static extern IntPtr GetStdHandle(uint handleId);
 
+    internal const int ERROR_ALREADY_EXISTS = 0xB7;
+
+    internal static bool IsMutexPresent(string name)
+    {
+        try
+        {
+            using var mutex = new System.Threading.Mutex(false, name);
+            return Marshal.GetLastWin32Error() == ERROR_ALREADY_EXISTS;
```

Comment:
@SeeminglyScience astutely pointed out this is bad, as we're not exactly making a system call when gettingn the mutex. It should be `TryOpenExisting`.